src/execution/mod.rs

```diff
@@ -78,10 +80,22 @@ pub(crate) fn create_instance(limit: u64, module: &impl MassaModule) -> Result<I
     let engine = Universal::new(compiler_config).features(FEATURES).engine();
     let store = Store::new_with_tunables(&engine, tunables);
 
-    Ok(Instance::new(
-        &Module::new(&store, module.get_bytecode())?,
+    match Instance::new(
+        &Module::new(&store, &module.get_bytecode())?,
         &module.resolver(&store),
-    )?)
+    ) {
+        Ok(i) => Ok(i),
+        Err(err) => {
+            if let InstantiationError::Start(ref e) = err {
+                if let Some(trap) = e.clone().to_trap() {
+                    if trap == TrapCode::UnreachableCodeReached && e.trace().len() == 0 {
```

Comment:
Are we sure this always means that we are out of gas ?

Reply:
We made multiple tests and it happens only in this case. We also checked in the code of wasmer and in the code of the middle the only place they add an unreachable at initialization is in metering in this case.

As we are before creating the instance we can't call `remaning_points` to be sure so it's our only solution.

Reply:
May be, could add a small comment to explain what we did here

Reply:
Done.